File: app/src/pitstop/java/com/pitstop/observer/BluetoothConnectionObserver.java
package com.pitstop.observer;

import com.pitstop.bluetooth.dataPackages.PidPackage;
import com.pitstop.models.ReadyDevice;

/**
 * Created by Karol Zdebel on 6/28/2017.
 */

public interface BluetoothConnectionObserver extends Observer{

    //Searching for bluetooth device associated with user car
    void onSearchingForDevice();

    //Device is ready to be interacted with
    void onDeviceReady(ReadyDevice readyDevice);

    //Device that was previously ready now disconnected and can no longer be interacted with
    void onDeviceDisconnected();

    //Device is connected and now being verified
    void onDeviceVerifying();

    //Device has been verified and is now syncing rtc time
    void onDeviceSyncing();

    //Received list of pids available for retrieval
    void onGotSuportedPIDs(String value);

    //A device from the list of devices has been picked and is being connected to
    void onConnectingToDevice();

    //Invoked once a list of devices is gathered after the discovery process
    void onFoundDevices();

    //Pids received from the vehicle
    void onGotPid(PidPackage pidPackage);

}

File: app/src/pitstop/java/com/pitstop/observer/BluetoothConnectionObserver.kt
package com.pitstop.observer

import com.continental.rvd.mobile_sdk.BindingQuestion
import com.continental.rvd.mobile_sdk.internal.api.binding.model.Error
import com.pitstop.bluetooth.dataPackages.PidPackage
import com.pitstop.models.ReadyDevice

/**
 * Created by Karol Zdebel on 6/28/2017.
 */

interface BluetoothConnectionObserver : Observer {

    //Searching for bluetooth device associated with user car
    fun onSearchingForDevice()

    //Device is ready to be interacted with
    fun onDeviceReady(readyDevice: ReadyDevice)

    //Device that was previously ready now disconnected and can no longer be interacted with
    fun onDeviceDisconnected()

    //Device is connected and now being verified
    fun onDeviceVerifying()

    //Device has been verified and is now syncing rtc time
    fun onDeviceSyncing()

    fun onGotSuportedPIDs(value: String)

    fun onConnectingToDevice()

    fun onFoundDevices()

    fun onGotPid(pidPackage: PidPackage)

    fun onBindingRequired()

    fun onBindingQuestionPrompted(question: BindingQuestion)

    fun onBindingProgress(progress: Float)

    fun onBindingFinished()

    fun onBindingError(err: Error)

    fun onFirmwareInstallationRequired()

    fun onFirmwareInstallationProgress(progress: Float)

    fun onFirmwareInstallationFinished()

    fun onFirmwareInstallationError(err: Error)
}
